assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -271,22 +274,21 @@ const baseActions = {
 					.select( MODULES_ANALYTICS_4 )
 					.getWebDataStreams( propertyID );
 
-				if ( webdatastreams && webdatastreams.length > 0 ) {
+				if ( webdatastreams?.length ) {
 					webdatastream = webdatastreams[ 0 ];
 				}
 			}
 
 			if ( webdatastream ) {
-				registry
-					.dispatch( MODULES_ANALYTICS_4 )
-					.setWebDataStreamID( webdatastream._id );
-				registry
-					.dispatch( MODULES_ANALYTICS_4 )
-					.updateSettingsForMeasurementID(
-						// eslint-disable-next-line sitekit/acronym-case
-						webdatastream.webStreamData.measurementId
-					);
+				setWebDataStreamID( webdatastream._id );
+				updateSettingsForMeasurementID(
+					// eslint-disable-next-line sitekit/acronym-case
+					webdatastream.webStreamData.measurementId
+				);
+				return;
 			}
+			// At this point there is no web data stream to set.
+			setWebDataStreamID( WEBDATASTREAM_CREATE );
```

Comment:
Stylistic preference – added return above hence the added comment to avoid the unnecessary `else`. We should avoid making this function any longer 😄